File: ocaml/stats.ml
(* Function to calculate mean *)
let mean numbers =
  let sum = List.fold_left ( + ) 0 numbers in
  float_of_int sum /. float_of_int (List.length numbers)

(* Function to calculate median *)
let median numbers =
  let sorted = List.sort compare numbers in
  let len = List.length sorted in
  if len mod 2 = 0 then
    let mid1 = List.nth sorted ((len / 2) - 1) in
    let mid2 = List.nth sorted (len / 2) in
    (float_of_int mid1 +. float_of_int mid2) /. 2.0
  else
    float_of_int (List.nth sorted (len / 2))

(* Function to calculate mode *)
let mode numbers =
  let rec count_occurrences acc = function
    | [] -> acc
    | x :: xs ->
      let current_count = List.assoc_opt x acc |> Option.value ~default:0 in
      count_occurrences ((x, current_count + 1) :: acc) xs
  in
  let counts = count_occurrences [] numbers in
  let max_count = List.fold_left (fun max (_, count) -> if count > max then count else max) 0 counts in
  List.fold_left
    (fun modes (num, count) -> if count = max_count then num :: modes else modes)
    [] counts

(* Main function to demonstrate usage *)
let () =
  let numbers = [1; 2; 3; 4; 5; 5; 6; 7; 8; 9] in
  Printf.printf "Numbers: ";
  List.iter (Printf.printf "%d ") numbers;
  Printf.printf "\n";
  
  Printf.printf "Mean: %.2f\n" (mean numbers);
  Printf.printf "Median: %.2f\n" (median numbers);
  
  Printf.printf "Mode: ";
  List.iter (Printf.printf "%d ") (mode numbers);
  Printf.printf "\n" 

File: python/stats.py
from typing import List
from collections import Counter

class StatisticsCalculator:
    def __init__(self, numbers: List[int]):
        self.numbers = numbers

    def calculate_mean(self) -> float:
        """Calculate the mean (average) of the numbers."""
        return sum(self.numbers) / len(self.numbers)

    def calculate_median(self) -> float:
        """Calculate the median of the numbers."""
        sorted_numbers = sorted(self.numbers)
        n = len(sorted_numbers)
        if n % 2 == 0:
            mid1 = sorted_numbers[n//2 - 1]
            mid2 = sorted_numbers[n//2]
            return (mid1 + mid2) / 2
        else:
            return sorted_numbers[n//2]

    def calculate_mode(self) -> List[int]:
        """Calculate the mode(s) of the numbers."""
        counter = Counter(self.numbers)
        max_count = max(counter.values())
        return [num for num, count in counter.items() if count == max_count]

    def print_statistics(self) -> None:
        """Print all statistics for the numbers."""
        print(f"Numbers: {' '.join(map(str, self.numbers))}")
        print(f"Mean: {self.calculate_mean():.2f}")
        print(f"Median: {self.calculate_median():.2f}")
        print(f"Mode: {' '.join(map(str, self.calculate_mode()))}")

def main():
    # Example usage
    numbers = [1, 2, 3, 4, 5, 5, 6, 7, 8, 9]
    calculator = StatisticsCalculator(numbers)
    calculator.print_statistics()

if __name__ == "__main__":
    main() 